ddl/job_table.go

```diff
@@ -496,14 +497,236 @@ func getJobsBySQL(sess *session, tbl, condition string) ([]*model.Job, error) {
 	return jobs, nil
 }
 
+// AddBackfillJobs adds the backfill jobs to the tidb_ddl_backfill table.
+func AddBackfillJobs(sess *session, backfillJobs []*BackfillJob) error {
+	sqlPrefix := "insert into mysql.tidb_ddl_backfill(section_id, job_id, ele_id, ele_key, store_id, type, exec_id, exec_lease, state, backfill_meta) values"
+	var sql string
+	// Do runInTxn for get StartTS.
+	return runInTxn(NewSession(sess), func(se *session) error {
+		txn, err := se.txn()
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		startTS := txn.StartTS()
+		for i, bj := range backfillJobs {
+			bj.Mate.StartTS = startTS
+			mateByte, err := bj.Mate.Encode()
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			if i == 0 {
+				sql = sqlPrefix + fmt.Sprintf("(%d, %d, %d, '%s', %d, %d, '%s', '%s', %d, '%s')",
+					bj.ID, bj.JobID, bj.EleID, bj.EleKey, bj.StoreID, bj.Tp, bj.InstanceID, bj.InstanceLease, bj.State, mateByte)
+				continue
+			}
+			sql += fmt.Sprintf(", (%d, %d, %d, '%s', %d, %d, '%s', '%s', %d, '%s')",
```

Comment:
For huge table, will this happened, when there are too many regions, finially make the txn memory limition get and txn failed.

Reply:
When we call it, we call it in batches.

Reply:
whether above failure will occur?

Reply:
No, because where we're calling it we're adding the shard function